service/history/historyEngine.go

```diff
@@ -2397,6 +2388,23 @@ func validateActivityScheduleAttributes(attributes *workflow.ScheduleActivityTas
 		}
 	}
 
+	validScheduleToClose := attributes.GetScheduleToCloseTimeoutSeconds() > 0
+	validScheduleToStart := attributes.GetScheduleToStartTimeoutSeconds() > 0
+	validStartToClose := attributes.GetStartToCloseTimeoutSeconds() > 0
+
+	if validScheduleToClose {
+		if !validScheduleToStart {
```

Comment:
we should do the overwrite only when the ScheduleToStartTimeoutSeconds is 0

Reply:
resolved